diff --git a/edu-services/gradebook-service/hibernate/src/java/org/sakaiproject/tool/gradebook/GradingEvents.java b/edu-services/gradebook-service/hibernate/src/java/org/sakaiproject/tool/gradebook/GradingEvents.java
index c7cc04a44b9..406ce689a29 100644
--- a/edu-services/gradebook-service/hibernate/src/java/org/sakaiproject/tool/gradebook/GradingEvents.java
+++ b/edu-services/gradebook-service/hibernate/src/java/org/sakaiproject/tool/gradebook/GradingEvents.java
@@ -27,11 +27,7 @@
  * @author <a href="mailto:jholtzman@berkeley.edu">Josh Holtzman</a>
  */
 public class GradingEvents implements Serializable {
-    protected Map<String, List<GradingEvent>> studentsToEventsMap;
-
-    public GradingEvents() {
-        studentsToEventsMap = new HashMap<>();
-    }
+    private Map<String, List<GradingEvent>> studentsToEventsMap = new HashMap<>();
 
     /**
      * Returns a list of grading events, which may be empty if none exist.
@@ -40,7 +36,11 @@ public GradingEvents() {
      * @return
      */
     public List<GradingEvent> getEvents(String studentId) {
-        return studentsToEventsMap.get(studentId);
+        List<GradingEvent> list = studentsToEventsMap.get(studentId);
+        if (list == null) {
+            list = new ArrayList<>();
+        }
+        return list;
     }
 
     public void addEvent(GradingEvent event) {
