diff --git a/gradebook/app/business/src/java/org/sakaiproject/tool/gradebook/business/impl/GradebookManagerHibernateImpl.java b/gradebook/app/business/src/java/org/sakaiproject/tool/gradebook/business/impl/GradebookManagerHibernateImpl.java
index e22a731dc86..b03f9613a6d 100644
--- a/gradebook/app/business/src/java/org/sakaiproject/tool/gradebook/business/impl/GradebookManagerHibernateImpl.java
+++ b/gradebook/app/business/src/java/org/sakaiproject/tool/gradebook/business/impl/GradebookManagerHibernateImpl.java
@@ -2183,12 +2183,10 @@
     @Override
     public List<GradebookAssignment> getAssignmentsWithNoCategory(final Long gradebookId, final String assignmentSort, final boolean assignAscending)
     {
-    	HibernateCallback<List<GradebookAssignment>> hc = session -> {
-            return session.createQuery(
-                    "from GradebookAssignment as asn where asn.gradebook.id = :id and asn.removed=false and asn.category is null").
-                    setLong(0, gradebookId.longValue()).
-                    list();
-        };
+		HibernateCallback<List<GradebookAssignment>> hc = session -> session
+				.createQuery("from GradebookAssignment as asn where asn.gradebook.id = :id and asn.removed=false and asn.category is null")
+				.setLong("id", gradebookId)
+				.list();
 
     	final List<GradebookAssignment> assignList = getHibernateTemplate().execute(hc);
     	if(assignmentSort != null) {
