diff --git kernel/api/src/main/java/org/sakaiproject/event/api/UsageSession.java b/kernel/api/src/main/java/org/sakaiproject/event/api/UsageSession.java
index 4c873bb..1b0e2ff 100644
--- kernel/api/src/main/java/org/sakaiproject/event/api/UsageSession.java
+++ kernel/api/src/main/java/org/sakaiproject/event/api/UsageSession.java
@@ -32,9 +32,15 @@
 {
 	/** String constants for major browser types */
 	public static final String MAC_IE = "Mac-InternetExplorer";
-
+	
+	/*
+	 * @deprecated  discontinued 2008
+	 */
 	public static final String MAC_NN = "Mac-NetscapeNavigator";
-
+	
+	/*
+	 * @deprecated  discontinued 2003
+	 */
 	public static final String MAC_CM = "Mac-Camino";
 
 	public static final String MAC_SF = "Mac-Safari";
@@ -43,11 +49,19 @@
 
 	public static final String WIN_IE = "Win-InternetExplorer";
 
+	/*
+	 * @deprecated  discontinued 2008
+	 */
 	public static final String WIN_NN = "Win-NetscapeNavigator";
 
 	public static final String WIN_MZ = "Win-Mozilla";
 	
 	public static final String LIN_MZ = "Lin-Mozilla";
+	
+	/*
+	 * Apple iOS - iPhone, iPad
+	 */
+	public static final String IOS_MZ = "iOS-Mozilla";
 
 	public static final String UNKNOWN = "UnknownBrowser";
 
diff --git kernel/kernel-impl/src/main/java/org/sakaiproject/event/impl/BaseUsageSession.java b/kernel/kernel-impl/src/main/java/org/sakaiproject/event/impl/BaseUsageSession.java
index 308ef52..26d5fa3 100644
--- kernel/kernel-impl/src/main/java/org/sakaiproject/event/impl/BaseUsageSession.java
+++ kernel/kernel-impl/src/main/java/org/sakaiproject/event/impl/BaseUsageSession.java
@@ -191,6 +191,19 @@ else if (agent.startsWith("Mozilla") && agent.indexOf("Linux") >= 0)
 		{
 			m_browserId = LIN_MZ;
 		}
+		//KNL-1490 Android => Linux
+		else if (agent.startsWith("Mozilla") && agent.indexOf("Android") >= 0)
+		{
+			m_browserId = LIN_MZ;
+		}
+		else if (agent.startsWith("Mozilla") && agent.indexOf("iPad") >= 0)
+		{
+			m_browserId = IOS_MZ;
+		}
+		else if (agent.startsWith("Mozilla") && agent.indexOf("iPhone") >= 0)
+		{
+			m_browserId = IOS_MZ;
+		}
 		else
 		{
 			m_browserId = UNKNOWN;
