diff --git a/kernel/kernel-impl/src/main/java/org/sakaiproject/user/impl/BaseUserDirectoryService.java b/kernel/kernel-impl/src/main/java/org/sakaiproject/user/impl/BaseUserDirectoryService.java
index 67fc68dc0d6..ef32ca26f0a 100644
--- a/kernel/kernel-impl/src/main/java/org/sakaiproject/user/impl/BaseUserDirectoryService.java
+++ b/kernel/kernel-impl/src/main/java/org/sakaiproject/user/impl/BaseUserDirectoryService.java
@@ -2521,7 +2521,7 @@ public Date getModifiedDate()
 		{
 			Date date = null;
 			if (m_lastModifiedInstant != null) {
-				date = new Date(m_lastModifiedInstant.getEpochSecond());
+				date = new Date(m_lastModifiedInstant.toEpochMilli());
 			}
 			return date;
 		}
