diff --git a/samigo/samigo-services/src/java/org/sakaiproject/tool/assessment/facade/AssessmentGradingFacadeQueries.java b/samigo/samigo-services/src/java/org/sakaiproject/tool/assessment/facade/AssessmentGradingFacadeQueries.java
index 7999827de60..77a00327592 100644
--- a/samigo/samigo-services/src/java/org/sakaiproject/tool/assessment/facade/AssessmentGradingFacadeQueries.java
+++ b/samigo/samigo-services/src/java/org/sakaiproject/tool/assessment/facade/AssessmentGradingFacadeQueries.java
@@ -1827,7 +1827,8 @@ public int getActualNumberRetake(final Long publishedAssessmentId, final String
         };
         List<Object[]> countList = getHibernateTemplate().execute(hcb);
         for (Object[] o : countList) {
-            actualNumberRetakeHash.put((Long) o[0], (Integer) o[1]);
+            Long l = (Long) o[1];
+            actualNumberRetakeHash.put((Long) o[0], l.intValue());
         }
         return actualNumberRetakeHash;
     }
