diff -ur /usr/local/src/vula_src/branches/vula-11.x/mailarchive//mailarchive-api/api/src/java/org/sakaiproject/mailarchive/api/MailArchiveMessage.java mailarchive//mailarchive-api/api/src/java/org/sakaiproject/mailarchive/api/MailArchiveMessage.java
--- /usr/local/src/vula_src/branches/vula-11.x/mailarchive//mailarchive-api/api/src/java/org/sakaiproject/mailarchive/api/MailArchiveMessage.java	2016-11-08 16:03:04.000000000 +0200
+++ mailarchive//mailarchive-api/api/src/java/org/sakaiproject/mailarchive/api/MailArchiveMessage.java	2017-03-13 15:58:23.000000000 +0200
@@ -54,4 +54,7 @@
 	 */
 	public String getFormattedBody();
 
+	/** Plain body */
+	public String getPlainBody();
+
 }
diff -ur /usr/local/src/vula_src/branches/vula-11.x/mailarchive//mailarchive-impl/impl/src/java/org/sakaiproject/mailarchive/impl/BaseMailArchiveService.java mailarchive//mailarchive-impl/impl/src/java/org/sakaiproject/mailarchive/impl/BaseMailArchiveService.java
--- /usr/local/src/vula_src/branches/vula-11.x/mailarchive//mailarchive-impl/impl/src/java/org/sakaiproject/mailarchive/impl/BaseMailArchiveService.java	2016-11-08 16:03:05.000000000 +0200
+++ mailarchive//mailarchive-impl/impl/src/java/org/sakaiproject/mailarchive/impl/BaseMailArchiveService.java	2017-03-13 15:57:29.000000000 +0200
@@ -1173,6 +1173,10 @@
 				
 		} // getHtmlBody
 
+		public String getPlainBody() {
+			return Web.encodeUrlsAsHtml( FormattedText.convertPlaintextToFormattedText(m_body) );
+		}
+
 		/**
 		 * Take all values from this object.
 		 * 
diff -ur /usr/local/src/vula_src/branches/vula-11.x/mailarchive//mailarchive-tool/tool/src/webapp/vm/mailarchive/chef_mailbox-view.vm mailarchive//mailarchive-tool/tool/src/webapp/vm/mailarchive/chef_mailbox-view.vm
--- /usr/local/src/vula_src/branches/vula-11.x/mailarchive//mailarchive-tool/tool/src/webapp/vm/mailarchive/chef_mailbox-view.vm	2016-11-08 16:03:11.000000000 +0200
+++ mailarchive//mailarchive-tool/tool/src/webapp/vm/mailarchive/chef_mailbox-view.vm	2017-03-13 15:58:54.000000000 +0200
@@ -53,7 +53,7 @@
 		#end
 		<tr>
 			<th>$tlang.getString("gen.bod2")</th>
-			<td>$validator.escapeHtmlFormattedText($email.FormattedBody)</td>
+			<td>$validator.escapeHtmlFormattedText($email.PlainBody)</td>
 		</tr>
 		#set ($attachments = $email.Header.Attachments)
 		#if (!$attachments.isEmpty())
