 
diff --git reference/library/src/webapp/js/lang-datepicker/lang-datepicker.js b/reference/library/src/webapp/js/lang-datepicker/lang-datepicker.js
index 626fdba..b9cea61 100644
--- reference/library/src/webapp/js/lang-datepicker/lang-datepicker.js
+++ reference/library/src/webapp/js/lang-datepicker/lang-datepicker.js
@@ -289,6 +289,24 @@
 showMonthAfterYear: false,
 yearSuffix: ''};
 $.datepicker.regional['en-NZ'] = {
+closeText: 'Done',
+prevText: 'Prev',
+nextText: 'Next',
+currentText: 'Today',
+monthNames: ['January','February','March','April','May','June',
+'July','August','September','October','November','December'],
+monthNamesShort: ['Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun',
+'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec'],
+dayNames: ['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday'],
+dayNamesShort: ['Sun', 'Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat'],
+dayNamesMin: ['Su','Mo','Tu','We','Th','Fr','Sa'],
+weekHeader: 'Wk',
+dateFormat: 'dd/mm/yy',
+firstDay: 1,
+isRTL: false,
+showMonthAfterYear: false,
+yearSuffix: ''};
+$.datepicker.regional['en-ZA'] = {
 closeText: 'Done',
 prevText: 'Prev',
 nextText: 'Next',
