diff -ur /usr/local/src/vula_src/branches/vula-12.x/emailtemplateservice/impl/logic/src/java/org/sakaiproject/emailtemplateservice/service/impl/EmailTemplateServiceImpl.java emailtemplateservice/impl/logic/src/java/org/sakaiproject/emailtemplateservice/service/impl/EmailTemplateServiceImpl.java
--- /usr/local/src/vula_src/branches/vula-12.x/emailtemplateservice/impl/logic/src/java/org/sakaiproject/emailtemplateservice/service/impl/EmailTemplateServiceImpl.java	2018-01-05 11:44:22.000000000 +0200
+++ emailtemplateservice/impl/logic/src/java/org/sakaiproject/emailtemplateservice/service/impl/EmailTemplateServiceImpl.java	2018-07-19 11:30:17.282657032 +0200
@@ -447,7 +447,10 @@
 		headers.add("Subject: " + rt.getRenderedSubject());
 		headers.add("Content-Type: multipart/alternative; boundary=\"" + MULTIPART_BOUNDARY + "\"");
 		headers.add("Mime-Version: 1.0");
-		headers.add("Precedence: bulk");
+
+		if (toAddress.size() > 1) {
+			headers.add("Precedence: bulk");
+		}
 
 		String body = message.toString();
 		log.debug("message body " + body);
