diff --git a/content-review/impl/turnitin/src/main/java/org/sakaiproject/contentreview/turnitin/TurnitinReviewServiceImpl.java b/content-review/impl/turnitin/src/main/java/org/sakaiproject/contentreview/turnitin/TurnitinReviewServiceImpl.java
index 0b116009468..bb94e27edf2 100644
--- a/content-review/impl/turnitin/src/main/java/org/sakaiproject/contentreview/turnitin/TurnitinReviewServiceImpl.java
+++ b/content-review/impl/turnitin/src/main/java/org/sakaiproject/contentreview/turnitin/TurnitinReviewServiceImpl.java
@@ -1007,13 +1007,14 @@ public void createAssignment(String siteId, String taskId, Map extraAsnnOpts)
 			throws SubmissionException, TransientSubmissionException {
 
 		// get the assignment reference
-		String taskTitle = "";
+		String rawTitle = "";
 		if (extraAsnnOpts.containsKey("title")) {
-			taskTitle = extraAsnnOpts.get("title").toString();
+			rawTitle = extraAsnnOpts.get("title").toString();
 		} else {
-			getAssignmentTitle(taskId);
+			rawTitle = getAssignmentTitle(taskId);
 		}
-		log.debug("Creating assignment for site: " + siteId + ", task: " + taskId + " tasktitle: " + taskTitle);
+		String taskTitle = rawTitle.replaceAll("[^\\w\\s]", "");
+		log.debug("Creating assignment for site: " + siteId + ", task: " + taskId + ", rawTitle: " + rawTitle + ", taskTitle: " + taskTitle);
 
 		SimpleDateFormat dform = ((SimpleDateFormat) DateFormat.getDateInstance());
 		dform.applyPattern(TURNITIN_DATETIME_FORMAT);
