diff --git a/kernel/kernel-impl/src/main/java/org/sakaiproject/memory/impl/EhcacheMemoryService.java b/kernel/kernel-impl/src/main/java/org/sakaiproject/memory/impl/EhcacheMemoryService.java
index d389af3ab87..cc2d3ec66d9 100644
--- a/kernel/kernel-impl/src/main/java/org/sakaiproject/memory/impl/EhcacheMemoryService.java
+++ b/kernel/kernel-impl/src/main/java/org/sakaiproject/memory/impl/EhcacheMemoryService.java
@@ -30,6 +30,7 @@
 
 import net.sf.ehcache.CacheManager;
 import net.sf.ehcache.Ehcache;
+import net.sf.ehcache.ObjectExistsException;
 import net.sf.ehcache.config.CacheConfiguration;
 import net.sf.ehcache.config.Configuration;
 
@@ -130,7 +131,12 @@ public Properties getProperties() {
 
     @Override
     public Cache getCache(String cacheName) {
-        return new EhcacheCache(makeEhcache(cacheName, null));
+	try {
+		return new EhcacheCache(makeEhcache(cacheName, null));
+	} catch (ObjectExistsException ex) {
+		log.debug("Cache {} already exists; possibly created by another thread", cacheName);
+		return null;
+	}
     }
 
     @Override
