diff --git samigo/samigo-app/pom.xml b/samigo/samigo-app/pom.xml
index 8a2ced45fac..b4de520129c 100644
--- samigo/samigo-app/pom.xml
+++ samigo/samigo-app/pom.xml
@@ -213,11 +213,6 @@
 	    <scope>runtime</scope>
         </dependency>
         <dependency>
-            <groupId>struts</groupId>
-            <artifactId>struts</artifactId>
-            <version>1.1</version>
-        </dependency>
-        <dependency>
             <groupId>org.sakaiproject.rsf</groupId>
             <artifactId>sakai-rsf-web-templates</artifactId>
             <version>${project.version}</version>
diff --git samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/questionpool/QuestionPoolBean.java b/samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/questionpool/QuestionPoolBean.java
index 2f075a66ca3..8a8207a802f 100644
--- samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/questionpool/QuestionPoolBean.java
+++ samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/questionpool/QuestionPoolBean.java
@@ -57,7 +57,6 @@
 import org.apache.poi.ss.usermodel.Row;
 import org.apache.poi.ss.usermodel.Sheet;
 import org.apache.poi.ss.usermodel.Workbook;
-import org.apache.struts.upload.FormFile;
 import org.osid.shared.SharedException;
 import org.sakaiproject.event.cover.EventTrackingService;
 import org.sakaiproject.tool.assessment.business.questionpool.QuestionPoolTreeImpl;
@@ -133,7 +132,6 @@
   private String[] destItems = {  }; // items to delete
   private String sourcePart = null; // copy all questions from part
   private String destPool="0"; // for Move Pool Destination
-  private FormFile filename; // for import /export
   private int htmlIdLevel; // pass this to javascript:collapseAll()
   private String questionType; // the question type to add
   private int parentPoolSize= 0; // the question type to add
@@ -1198,26 +1196,6 @@ public void setSelectedPools(String[] pPool)
   /**
    * DOCUMENTATION PENDING
    *
-   * @param file DOCUMENTATION PENDING
-   */
-  public void setFilename(FormFile file)
-  {
-    filename = file;
-  }
-
-  /**
-   * DOCUMENTATION PENDING
-   *
-   * @param file DOCUMENTATION PENDING
-   */
-  public FormFile getFilename()
-  {
-    return filename;
-  }
-
-  /**
-   * DOCUMENTATION PENDING
-   *
    * @param int DOCUMENTATION PENDING
    */
   public int getHtmlIdLevel()
