--- /usr/local/src/vula_src/branches/vula-11.x/samigo/samigo-services/src/java/org/sakaiproject/tool/assessment/facade/AssessmentGradingFacadeQueries.java	2017-07-15 05:06:22.000000000 +0200
+++ samigo/samigo-services/src/java/org/sakaiproject/tool/assessment/facade/AssessmentGradingFacadeQueries.java	2018-01-08 12:44:16.000000000 +0200
@@ -1337,11 +1337,12 @@
     int retryCount = persistenceHelper.getRetryCount();
     while (retryCount > 0){ 
       try {
-        /* for testing the catch block - daisyf 
-        if (retryCount >2)
-          throw new Exception("uncategorized SQLException for SQL []; SQL state [61000]; error code [60]; ORA-00060: deadlock detected while waiting for resource");
-	*/ 
-        getHibernateTemplate().saveOrUpdate((AssessmentGradingData)assessment);
+        if (assessment.getAssessmentGradingId() != null) {
+            getHibernateTemplate().merge((AssessmentGradingData) assessment);
+        }
+        else {
+            getHibernateTemplate().save((AssessmentGradingData) assessment);
+        }
         retryCount = 0;
       }
       catch (Exception e) {
@@ -1865,7 +1866,7 @@
     while (retryCount > 0){ 
       try {
           for (ItemGradingData itemGradingData : c) {
-              getHibernateTemplate().saveOrUpdate(itemGradingData);
+              getHibernateTemplate().merge(itemGradingData);
           }
         retryCount = 0;
       }
