diff -ur '--exclude=target' '--exclude=*properties*' assignment/impl/src/java/org/sakaiproject/assignment/impl/conversion/impl/AssignmentConversionServiceImpl.java assignment-patched//impl/src/java/org/sakaiproject/assignment/impl/conversion/impl/AssignmentConversionServiceImpl.java
--- assignment/impl/src/java/org/sakaiproject/assignment/impl/conversion/impl/AssignmentConversionServiceImpl.java	2018-01-29 14:24:59.000000000 +0200
+++ assignment-patched//impl/src/java/org/sakaiproject/assignment/impl/conversion/impl/AssignmentConversionServiceImpl.java	2018-01-29 14:03:25.000000000 +0200
@@ -31,6 +31,7 @@
 import javax.xml.stream.XMLStreamException;
 import javax.xml.stream.events.XMLEvent;
 
+import com.ctc.wstx.api.WstxInputProperties;
 import com.fasterxml.jackson.core.JsonParser;
 import com.fasterxml.jackson.core.JsonProcessingException;
 import com.fasterxml.jackson.databind.DeserializationContext;
@@ -78,7 +79,12 @@
             }
         });
 
-        xmlMapper = new XmlMapper();
+        XMLInputFactory xmlInputFactory = XMLInputFactory.newFactory();
+
+        xmlInputFactory.setProperty(WstxInputProperties.P_MAX_ATTRIBUTES_PER_ELEMENT, 8000);
+        xmlInputFactory.setProperty(WstxInputProperties.P_MAX_ATTRIBUTE_SIZE, 2097152);
+
+        xmlMapper = new XmlMapper(xmlInputFactory);
         xmlMapper.registerModule(new Jdk8Module());
         xmlMapper.registerModule(module);
     }
@@ -110,7 +116,7 @@
             try {
                 return xmlMapper.readValue(xml, clazz);
             } catch (IOException ioe) {
-                log.warn("deserialization failed for xml: {}\n{}", xml.substring(0, Math.min(xml.length(), 200)), ioe.getMessage());
+                log.warn("deserialization failed for xml: {}\n{}", xml.substring(0, Math.min(xml.length(), 200)), ioe.getMessage(), ioe);
             }
         }
         return null;
diff -ur '--exclude=target' '--exclude=*properties*' assignment/impl/src/java/org/sakaiproject/assignment/impl/conversion/impl/AssignmentDataProvider11.java assignment-patched//impl/src/java/org/sakaiproject/assignment/impl/conversion/impl/AssignmentDataProvider11.java
--- assignment/impl/src/java/org/sakaiproject/assignment/impl/conversion/impl/AssignmentDataProvider11.java	2018-01-29 14:24:59.000000000 +0200
+++ assignment-patched//impl/src/java/org/sakaiproject/assignment/impl/conversion/impl/AssignmentDataProvider11.java	2018-01-29 12:04:19.000000000 +0200
@@ -21,7 +21,7 @@
         List<String> list;
         try {
             list = sessionFactory.getCurrentSession()
-                    .createSQLQuery("SELECT ASSIGNMENT_ID FROM ASSIGNMENT_ASSIGNMENT")
+                    .createSQLQuery("SELECT ASSIGNMENT_ID FROM ASSIGNMENT_ASSIGNMENT inner join SAKAI_SITE ON ASSIGNMENT_ASSIGNMENT.CONTEXT = SAKAI_SITE.SITE_ID ORDER BY CREATEDON DESC")
                     .addScalar("ASSIGNMENT_ID", StringType.INSTANCE)
                     .list();
 
