diff --git a/portal/portal-render-engine-impl/impl/src/webapp/vm/morpheus/includeLoginNav.vm b/portal/portal-render-engine-impl/impl/src/webapp/vm/morpheus/includeLoginNav.vm
index 8d6ada1..79cc8e4 100644
--- a/portal/portal-render-engine-impl/impl/src/webapp/vm/morpheus/includeLoginNav.vm
+++ b/portal/portal-render-engine-impl/impl/src/webapp/vm/morpheus/includeLoginNav.vm
@@ -76,7 +76,7 @@
 
                                         <div id="loginUser" class="has-avatar Mrphs-userNav__submenuitem--userlink #if (${site.isCurrentSite}) current-site #end">
                                             <a href="javascript:void(0);" role="button" aria-haspopup="true" class="Mrphs-userNav__drop-btn Mrphs-userNav__submenuitem--username">
-                                                <span class="Mrphs-userNav__submenuitem--profilepicture" style="background-image:url(/direct/profile/${loginUserId}/image/thumb)"></span>
+                                                <span class="Mrphs-userNav__submenuitem--profilepicture" style="background-image:url(/direct/profileClassic/${loginUserDispId}/image?_=${profileLastUpdate})"></span>
                                                 <span class="textlink">${loginUserFirstName}</span>
                                             </a>
                                         </div>
@@ -99,7 +99,7 @@
                                         #if (${tabsSites.mrphs_profileToolUrl})
                                             <div class="has-avatar">
                                                 <a role="menuitem" class="Mrphs-userNav__submenuitem--profilelink" href="${tabsSites.mrphs_profileToolUrl}" aria-haspopup="dialog">
-                                                    <span class="Mrphs-userNav__submenuitem--profilepicture Mrphs-userNav__pic-changer" style="background-image:url(/direct/profile/${loginUserId}/image/thumb)"></span>
+                                                    <span class="Mrphs-userNav__submenuitem--profilepicture" style="background-image:url(/direct/profileClassic/${loginUserDispId}/image?_=${profileLastUpdate})"></span>
                                                     <span class="sr-only">${rloader.pic_changer_title}</span>
                                                 </a>
                                             </div>
