diff -ur /usr/local/src/vula_src/branches/vula-12.x/chat/chat-impl/impl/src/java/org/sakaiproject/chat2/model/impl/ChatManagerImpl.java chat/chat-impl/impl/src/java/org/sakaiproject/chat2/model/impl/ChatManagerImpl.java
--- /usr/local/src/vula_src/branches/vula-12.x/chat/chat-impl/impl/src/java/org/sakaiproject/chat2/model/impl/ChatManagerImpl.java      2018-01-05 11:44:22.000000000 +0200
+++ chat/chat-impl/impl/src/java/org/sakaiproject/chat2/model/impl/ChatManagerImpl.java 2018-07-18 14:53:32.724807917 +0200
@@ -109,7 +109,7 @@
 @Slf4j
 public class ChatManagerImpl extends HibernateDaoSupport implements ChatManager, Receiver {

-    private int messagesMax = 100;
+    private int messagesMax = 1000;

     @Getter @Setter private ChatChannel defaultChannelSettings;


