--- /usr/local/src/vula_src/branches/vula-12.x/dav/dav/src/java/org/sakaiproject/dav/DavServlet.java	2018-01-05 11:44:22.000000000 +0200
+++ dav/dav/src/java/org/sakaiproject/dav/DavServlet.java	2018-01-22 15:23:13.000000000 +0200
@@ -1114,6 +1114,7 @@
 						&& !UsageSessionService.login(a, req, UsageSessionService.EVENT_LOGIN_DAV))
 				{
 					// login failed
+					doAuthFailPause();
 					res.addHeader("WWW-Authenticate","Basic realm=\"DAV\"");
 					res.sendError(401);
 					return;
@@ -1122,6 +1123,7 @@
 			catch (AuthenticationException ex)
 			{
 				// not authenticated
+				doAuthFailPause();
 				res.addHeader("WWW-Authenticate","Basic realm=\"DAV\"");
 				res.sendError(401);
 				return;
@@ -1130,6 +1132,7 @@
 		else
 		{
 			// user name missing, so can't authenticate
+			doAuthFailPause();
 			res.addHeader("WWW-Authenticate","Basic realm=\"DAV\"");
 			res.sendError(401);
 			return;
@@ -1156,6 +1159,20 @@
 		}
 	}
 
+
+       /**
+        * Pause the request on auth failure
+        */
+       private void doAuthFailPause() {
+               log.info("Pausing DAV request after failed authentication");
+               try {
+                       Thread.sleep(50);
+               } catch (InterruptedException e) {
+                       ///nothing to do here ...
+                       e.printStackTrace();
+               }
+       }
+
 	/** log a request processed */
 	public void log(HttpServletRequest req, SakaidavServletInfo info)
 	{
@@ -1183,6 +1200,7 @@
 		if (remoteUser == null)
 		{
 			if (log.isDebugEnabled()) log.debug("SAKAIDAV Requires Authorization");
+			doAuthFailPause();
 			resp.sendError(HttpServletResponse.SC_UNAUTHORIZED);
 			return;
 		}
