--- /usr/local/src/vula_src/branches/vula-12.x/rwiki/rwiki-tool/tool/src/java/uk/ac/cam/caret/sakai/rwiki/tool/bean/DiffBean.java	2018-01-05 11:39:22.000000000 +0200
+++ rwiki/rwiki-tool/tool/src/java/uk/ac/cam/caret/sakai/rwiki/tool/bean/DiffBean.java	2019-01-08 11:55:16.351844015 +0200
@@ -104,13 +104,13 @@
 
 	private void setLeftVersionNumber(String versionString)
 	{
-		if (versionString != null && !"".equals(versionString))
-		{
-			leftVersionNumber = Integer.parseInt(versionString);
-		}
-
 		try
 		{
+			if (versionString != null && !"".equals(versionString))
+			{
+				leftVersionNumber = Integer.parseInt(versionString);
+			}
+
 			if (rwo.getRevision().intValue() == leftVersionNumber)
 			{
 				left = rwo;
@@ -123,32 +123,33 @@
 		}
 		catch (IllegalArgumentException e)
 		{
-			throw new IllegalArgumentException(
-					"Invalid version number for left revision");
+			// Default to current revision
+			left = rwo;
 		}
-
 	}
 
 	private void setRightVersionNumber(String versionString)
 	{
 		if (versionString != null && !"".equals(versionString))
 		{
-			rightVersionNumber = Integer.parseInt(versionString);
-
 			try
 			{
+				rightVersionNumber = Integer.parseInt(versionString);
+
 				if (rwo.getRevision().intValue() == rightVersionNumber)
 				{
 					right = rwo;
 				}
 				else
+				{
 					right = rwikiObjectService.getRWikiHistoryObject(rwo,
 							rightVersionNumber);
+				}
 			}
 			catch (IllegalArgumentException e)
 			{
-				throw new IllegalArgumentException(
-						"Invalid version number for right revision");
+				// Default to current revision
+				right = rwo;
 			}
 		}
 		else
@@ -218,6 +219,12 @@
 	{
 		if (db == null)
 		{
+			if (left == null) {
+				left = rwo;
+			}
+			if (right == null) {
+				right = rwo;
+			}
 			db = new GenericDiffBean(left.getContent(), right.getContent());
 		}
 		return db;
