diff --git a/portal/portal-impl/impl/src/java/org/sakaiproject/portal/charon/SkinnableCharonPortal.java b/portal/portal-impl/impl/src/java/org/sakaiproject/portal/charon/SkinnableCharonPortal.java
index 8a9273d..0b04146 100644
--- a/portal/portal-impl/impl/src/java/org/sakaiproject/portal/charon/SkinnableCharonPortal.java
+++ b/portal/portal-impl/impl/src/java/org/sakaiproject/portal/charon/SkinnableCharonPortal.java
@@ -1073,6 +1073,11 @@ public class SkinnableCharonPortal extends HttpServlet implements Portal
         // rcontext.put("sitHelp", Web.escapeHtml(rb.getString("sit_help")));
         // rcontext.put("sitReset", Web.escapeHtml(rb.getString("sit_reset")));
         
+                // Get timestamp of user's last profile update to invalidate old profile pictures in the browser cache
+                Session s = SessionManager.getCurrentSession();
+                String profileLastUpdate = (String) s.getAttribute("profileLastUpdate");
+                rcontext.put("profileLastUpdate", StringUtils.isBlank(profileLastUpdate) ? Long.toString(s.getCreationTime()/1000L) : profileLastUpdate);
+                
         //SAK-29457 Add warning about cookie use
         String cookieNoticeText = rloader.getFormattedMessage("cookie_notice_text", ServerConfigurationService.getString("portal.cookie.policy.warning.url","/library/content/cookie_policy.html"));
         rcontext.put("cookieNoticeEnabled", ServerConfigurationService.getBoolean("portal.cookie.policy.warning.enabled",false));

