diff --git a/login/login-render-engine-impl/impl/src/webapp/vm/defaultskin/choice.vm b/login/login-render-engine-impl/impl/src/webapp/vm/defaultskin/choice.vm
index 22e468d..86a3fb1 100644
--- a/login/login-render-engine-impl/impl/src/webapp/vm/defaultskin/choice.vm
+++ b/login/login-render-engine-impl/impl/src/webapp/vm/defaultskin/choice.vm
@@ -14,10 +14,10 @@
 
 		<!-- choice.vm -->
 		<div id="Mrphs-choice">
-			<h1>${loginRequired}</h1>
+			<h1>${choiceRequired}</h1>
 			<aside class="logo"></aside>
-			<p class="msg">${choiceRequired}</p>
-			<fieldset>
+			<!--p class="msg">${choiceRequired}</p-->
+			<div>
 				<a class="link" href="${containerLoginUrl}" title="${loginTitle}">
 					#if("$containerLoginChoiceIcon" != "")
 						<img src="${containerLoginChoiceIcon}" alt="${containerLoginChoiceText}" />
@@ -30,7 +30,7 @@
 					#end
 					${xloginChoiceText}
 				</a>
-			</fieldset>
+			</div>
 		</div>
 		<!-- end choice.vm -->
 				
