Index: basiclti/basiclti-portlet/src/java/org/sakaiproject/portlets/IMSBLTIPortlet.java
===================================================================
--- basiclti/basiclti-portlet/src/java/org/sakaiproject/portlets/IMSBLTIPortlet.java
+++ basiclti/basiclti-portlet/src/java/org/sakaiproject/portlets/IMSBLTIPortlet.java
@@ -283,6 +283,9 @@ public class IMSBLTIPortlet extends GenericPortlet {
 					text.append("      document.getElementById(idval).height = height;\n");
 					text.append("      console.log('Reveived lti.frameResize height='+height);\n");
 					text.append("    }\n");
+					text.append("    else if ( message.subject == 'lti.pageRefresh' ) {\n");
+					text.append("      location.reload(true);\n");
+					text.append("    }\n");
 					text.append("  } catch (error) {\n");
 					text.append("   console.log('lti.frameResize of '+idval+' failed height='+height);\n");
 					text.append("   console.log(e.data);\n");
