Index: edu-services/gradebook-service/impl/src/java/org/sakaiproject/component/gradebook/GradebookServiceHibernateImpl.java
===================================================================
--- edu-services/gradebook-service/impl/src/java/org/sakaiproject/component/gradebook/GradebookServiceHibernateImpl.java	(revision 320551)
+++ edu-services/gradebook-service/impl/src/java/org/sakaiproject/component/gradebook/GradebookServiceHibernateImpl.java	(working copy)
@@ -1319,9 +1319,10 @@
   				return new ArrayList();
   			} else {
   				Query q = session.createQuery("from AssignmentGradeRecord as agr where agr.gradableObject.removed=false and " +
-  				"agr.gradableObject.gradebook.id=:gradebookId order by agr.pointsEarned");
+				"agr.gradableObject.gradebook.id=:gradebookId and agr.studentId in (:studentId) order by agr.pointsEarned");
   				q.setLong("gradebookId", gradebookId.longValue());
-  				return filterGradeRecordsByStudents(q.list(), studentUids);
+				q.setParameterList("studentId", studentUids);
+  				return q.list();
   			}
   		}
   	};
