--- /usr/local/src/vula_src/branches/vula-12.x/rwiki/rwiki-tool/tool/src/java/uk/ac/cam/caret/sakai/rwiki/tool/RWikiServlet.java	2018-01-05 11:44:23.000000000 +0200
+++ rwiki/rwiki-tool/tool/src/java/uk/ac/cam/caret/sakai/rwiki/tool/RWikiServlet.java	2019-01-10 10:56:43.596093463 +0200
@@ -39,6 +39,7 @@
 import org.sakaiproject.tool.api.ToolSession;
 import org.sakaiproject.tool.cover.SessionManager;
 
+import uk.ac.cam.caret.sakai.rwiki.service.exception.PermissionException;
 import uk.ac.cam.caret.sakai.rwiki.tool.api.HttpCommand;
 import uk.ac.cam.caret.sakai.rwiki.tool.bean.PrePopulateBean;
 import uk.ac.cam.caret.sakai.rwiki.tool.bean.ViewBean;
@@ -194,12 +195,14 @@
 		RequestScopeSuperBean rssb = RequestScopeSuperBean.createAndAttach(
 				request, wac);
 		
-		
-		
 
 		PrePopulateBean ppBean = rssb.getPrePopulateBean();
 
-		ppBean.doPrepopulate();
+		try {
+			ppBean.doPrepopulate();
+		} catch (PermissionException pe) {
+			log.debug("No permission to create subspace default pages");
+		}
 	}
 
 	public void addWikiStylesheet(HttpServletRequest request)
--- /usr/local/src/vula_src/branches/vula-12.x/rwiki/rwiki-tool/tool/src/java/uk/ac/cam/caret/sakai/rwiki/tool/service/impl/PopulateServiceImpl.java	2018-01-05 11:44:23.000000000 +0200
+++ rwiki/rwiki-tool/tool/src/java/uk/ac/cam/caret/sakai/rwiki/tool/service/impl/PopulateServiceImpl.java	2019-01-10 11:03:56.546242367 +0200
@@ -29,6 +29,7 @@
 import java.util.List;
 
 import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang.StringUtils;
 
 import org.sakaiproject.component.api.ComponentManager;
 import org.sakaiproject.event.cover.EventTrackingService;
@@ -39,6 +40,7 @@
 
 import uk.ac.cam.caret.sakai.rwiki.service.api.PageLinkRenderer;
 import uk.ac.cam.caret.sakai.rwiki.service.api.RenderService;
+import uk.ac.cam.caret.sakai.rwiki.service.api.RWikiSecurityService;
 import uk.ac.cam.caret.sakai.rwiki.service.api.dao.RWikiCurrentObjectDao;
 import uk.ac.cam.caret.sakai.rwiki.service.api.model.RWikiCurrentObject;
 import uk.ac.cam.caret.sakai.rwiki.service.exception.PermissionException;
@@ -56,6 +58,8 @@
 
 	private RWikiCurrentObjectDao dao;
 
+	private RWikiSecurityService securityService;
+
 	private RenderService renderService = null;
 
 	private SiteService siteService = null;
@@ -67,6 +71,7 @@
 
 		renderService = (RenderService) load(cm, RenderService.class.getName());
 		siteService = (SiteService) load(cm, SiteService.class.getName());
+		securityService = (RWikiSecurityService) load(cm, RWikiSecurityService.class.getName());
 
 		for (Iterator i = seedPages.iterator(); i.hasNext();)
 		{
@@ -132,20 +137,22 @@
 		}
 		catch (Exception e)
 		{
-			log
-					.warn("Cant find who created this site, defaulting to current user for prepopulate ownership :"
-							+ owner);
+			log.warn("Cant find who created this site, defaulting to current user for prepopulate ownership: {}", owner);
 		}
 		if (s == null)
 		{
-			log
-					.error("Cant Locate current site, will populate only global pages with no restrictions");
+			log.error("Cant Locate current site, will populate only global pages with no restrictions");
 		}
-		if (log.isDebugEnabled())
-		{
-			log.debug("Populating space: " + space);
+
+		boolean subspace = (StringUtils.countMatches(space, "/") > 2);
+
+		if (subspace && !securityService.checkCreatePermission(group)) {
+			throw new PermissionException("Not authorized to create subspaces in this site");
 		}
 
+		log.debug("Populating space: {} user: {} owner: {} group: {} subspace: {}",
+			space, user, owner, group, subspace);
+
 		for (Iterator i = seedPages.iterator(); i.hasNext();)
 		{
 
