diff --git a/portal/portal-render-engine-impl/impl/src/webapp/vm/morpheus/includeLoginNav.vm b/portal/portal-render-engine-impl/impl/src/webapp/vm/morpheus/includeLoginNav.vm
index 8d6ada1..2e48bab 100644
--- a/portal/portal-render-engine-impl/impl/src/webapp/vm/morpheus/includeLoginNav.vm
+++ b/portal/portal-render-engine-impl/impl/src/webapp/vm/morpheus/includeLoginNav.vm
@@ -103,11 +103,6 @@
                                                     <span class="sr-only">${rloader.pic_changer_title}</span>
                                                 </a>
                                             </div>
-                                            <div class="Mrphs-userNav__submenuitem--profile">
-                                                <a role="menuitem" href="${tabsSites.mrphs_profileToolUrl}">
-                                                    <span>${rloader.sit_profile}</span>
-                                                </a>
-                                            </div>
                                         #end
                                     </div>
                                     <div class="Mrphs-userNav__submenuitem--fullname-and-id">
@@ -120,11 +115,20 @@
                                     </div>
                                 </li>
 
-                                <li class="Mrphs-userNav__submenuitem Mrphs-userNav__submenuitem-indented">
+                                #if (${tabsSites.mrphs_profileToolUrl})
+                                <li class="Mrphs-userNav__submenuitem Mrphs-userNav__submenuitem-indented Mrphs-userNav__submenuitem--profile">
+                                    <a role="menuitem" href="${tabsSites.mrphs_profileToolUrl}">
+                                        <span>${rloader.sit_profile}</span>
+                                    </a>
+                                </li>
+                                #end
+
+                                <!--li class="Mrphs-userNav__submenuitem Mrphs-userNav__submenuitem-indented">
                                     <a role="menuitem" href="javascript:;" id="Mrphs-userNav__submenuitem--connections" aria-haspopup="dialog">
                                         <span>${rloader.sit_connections}</span>
                                     </a>
                                 </li>
+                                </li-->
 
                                     #if (${tabsSites.calendarToolUrl})
             
