Index: dav/dav/src/java/org/sakaiproject/dav/DavServlet.java
===================================================================
--- dav/dav/src/java/org/sakaiproject/dav/DavServlet.java	(revision 310399)
+++ dav/dav/src/java/org/sakaiproject/dav/DavServlet.java	(working copy)
@@ -1099,6 +1099,7 @@
 						&& !UsageSessionService.login(a, req, UsageSessionService.EVENT_LOGIN_DAV))
 				{
 					// login failed
+					doAuthFailPause();
 					res.addHeader("WWW-Authenticate","Basic realm=\"DAV\"");
 					res.sendError(401);
 					return;
@@ -1107,6 +1108,7 @@
 			catch (AuthenticationException ex)
 			{
 				// not authenticated
+				doAuthFailPause();
 				res.addHeader("WWW-Authenticate","Basic realm=\"DAV\"");
 				res.sendError(401);
 				return;
@@ -1115,6 +1117,7 @@
 		else
 		{
 			// user name missing, so can't authenticate
+			doAuthFailPause();
 			res.addHeader("WWW-Authenticate","Basic realm=\"DAV\"");
 			res.sendError(401);
 			return;
@@ -1141,6 +1144,20 @@
 		}
 	}
 
+
+       /**
+        * Pause the request on auth failure
+        */
+       private void doAuthFailPause() {
+               M_log.info("Pausing DAV request after failed authentication");
+               try {
+                       Thread.sleep(50);
+               } catch (InterruptedException e) {
+                       ///nothing to do here ...
+                       e.printStackTrace();
+               }
+       }
+
 	/** log a request processed */
 	public void log(HttpServletRequest req, SakaidavServletInfo info)
 	{
@@ -1168,6 +1185,7 @@
 		if (remoteUser == null)
 		{
 			if (M_log.isDebugEnabled()) M_log.debug("SAKAIDAV Requires Authorization");
+			doAuthFailPause();
 			resp.sendError(HttpServletResponse.SC_UNAUTHORIZED);
 			return;
 		}
