Index: assignment/assignment-impl/impl/src/java/org/sakaiproject/assignment/impl/BaseAssignmentService.java
===================================================================
--- assignment/assignment-impl/impl/src/java/org/sakaiproject/assignment/impl/BaseAssignmentService.java	(revision 311513)
+++ assignment/assignment-impl/impl/src/java/org/sakaiproject/assignment/impl/BaseAssignmentService.java	(working copy)
@@ -9947,7 +9983,7 @@
 			else
 			{
 				//we may have already retrived this one
-				if (m_reviewScore != null && m_reviewScore > -1) {
+				if (m_reviewScore != null && m_reviewScore > 0 && !"Error".equals(m_reviewReport) ) {
 					M_log.debug("returning stored value of " + m_reviewScore);
 					return m_reviewScore.intValue();
 				}
