diff --git a/portal/portal-render-engine-impl/impl/src/webapp/vm/morpheus/includeLoginNav.vm b/portal/portal-render-engine-impl/impl/src/webapp/vm/morpheus/includeLoginNav.vm
index cd31cc6..f5d25c7 100644
--- a/portal/portal-render-engine-impl/impl/src/webapp/vm/morpheus/includeLoginNav.vm
+++ b/portal/portal-render-engine-impl/impl/src/webapp/vm/morpheus/includeLoginNav.vm
@@ -157,6 +157,26 @@
 
                                     #end ## END of IF (${tabsSites.tutorial})
 
+                                    ##Link for password changing
+                                        <li class="Mrphs-userNav__submenuitem Mrphs-userNav__submenuitem-indented">
+                                    #if (${passwordURL})
+                                            <a role="menuitem" href="${passwordURL}" target="_blank" class="Mrphs-userNav__submenuitem--prefs">
+                                                <span>Change Password</span>
+                                            </a>
+                                    #else
+                                            <a role="menuitem" href="${tabsSites.mrphs_profileToolUrl}" class="Mrphs-userNav__submenuitem--prefs">   ##TODO: determine if tabsSites is accessible in SkinnableCharonPortal.java
+                                                <span>Change Password</span>
+                                            </a>
+                                    #end ## END of IF (${passwordURL})
+                                        </li>
+                                    #if (${peoplesoftURL})
+                                        <li class="Mrphs-userNav__submenuitem Mrphs-userNav__submenuitem-indented Mrphs-userData">
+                                            <a role="menuitem" href="${peoplesoftURL}" target="_blank" class="Mrphs-userNav__submenuitem--prefs">
+                                                <span>PeopleSoft</span>
+                                            </a>
+                                        </li>
+                                    #end  ## END of IF (${peoplesoftURL})                                    
+
                             #end ## END of IF (${site.isMyWorkspace})
 
                         #end ## END of FOREACH ( $site in $tabsSites.tabsSites )
