diff -ur /usr/local/src/vula_src/branches/vula-12.x/kernel/kernel-util/src/main/java/org/sakaiproject/util/BasicAuth.java kernel/kernel-util/src/main/java/org/sakaiproject/util/BasicAuth.java
--- /usr/local/src/vula_src/branches/vula-12.x/kernel/kernel-util/src/main/java/org/sakaiproject/util/BasicAuth.java	2018-01-05 11:44:22.000000000 +0200
+++ kernel/kernel-util/src/main/java/org/sakaiproject/util/BasicAuth.java	2018-09-08 10:15:00.777417742 +0200
@@ -213,7 +213,12 @@
 		
 			String auth;
 			auth = req.getHeader("Authorization");
-			
+
+			if (auth == null) {
+				// No basic auth header, so nothing to do
+				return false;
+			}
+
 			Evidence e = null;
 			try {
 				if (auth != null) {
Only in kernel/kernel-util/: target
