--- /usr/local/src/vula_src/branches/vula-11.x/reference/library/src/webapp/editor/ckeditor.launch.js	2016-12-20 05:24:44.000000000 +0200
+++ ckeditor.launch.js	2017-01-02 14:20:15.000000000 +0200
@@ -27,7 +27,7 @@
 sakai.editor.editors.ckeditor = sakai.editor.editors.ckeditor || {} ;
 
 //get path of directory ckeditor 
-var basePath = "/library/editor/ckextraplugins/"; 
+var basePath = "https://devslscle001.uct.ac.za/library/editor/ckextraplugins/"; 
 var webJars = "/library/webjars/"
 
 // Please note that no more parameters should be added to this signature.
@@ -160,7 +160,8 @@
             '/',
             ['Styles','Format','Font','FontSize'],
             ['TextColor','BGColor'],
-            ['Maximize', 'ShowBlocks']
+            ['Maximize', 'ShowBlocks'], 
+            ['ckeditor_wiris_formulaEditor']
         ],
         toolbar: 'Full',
         resize_dir: 'both',
@@ -200,6 +201,7 @@
             CKEDITOR.plugins.addExternal('autosave',webJars+'autosave/8541f541d9985cfd0859c7d8eb6be404afe95a2d/', 'plugin.js');
             CKEDITOR.plugins.addExternal('notification',basePath+'notification/', 'plugin.js');
             CKEDITOR.plugins.addExternal('fontawesome',basePath+'fontawesome/', 'plugin.js');
+            CKEDITOR.plugins.addExternal('ckeditor_wiris',basePath+'ckeditor_wiris/', 'plugin.js');
             /*
                To enable after the deadline uncomment these two lines and add atd-ckeditor to toolbar
                and to extraPlugins. This also needs extra stylesheets.
@@ -216,7 +218,7 @@
             //ckconfig.extraPlugins+="atd-ckeditor,";
             //ckconfig.contentsCss = [basePath+'atd-ckeditor/atd.css'];
 
-            ckconfig.extraPlugins+="image2,audiorecorder,movieplayer,wordcount,fmath_formula,autosave,fontawesome,notification";
+            ckconfig.extraPlugins+="image2,audiorecorder,movieplayer,wordcount,fmath_formula,autosave,fontawesome,notification,ckeditor_wiris";
 
             //SAK-29648
             ckconfig.contentsCss = [basePath+'/fontawesome/font-awesome/css/font-awesome.min.css'];
